Modules/_functoolsmodule.c

```diff
@@ -932,15 +932,31 @@ _functools_cmp_to_key_impl(PyObject *module, PyObject *mycmp)
 
 /* reduce (used to be a builtin) ********************************************/
 
-// Not converted to argument clinic, because of `args` in-place modification.
-// AC will affect performance.
+/*[clinic input]
+_functools.reduce
+
+    function as func: object
+    iterable as seq: object
+    initial as result: object(c_default="NULL") = _functools._initial_missing
+    /
+
+Apply a function of two arguments cumulatively to an iterable, from left to right.
+
+This efficiently reduces the iterable to a single value.  If initial is present,
+it is placed before the items of the iterable in the calculation, and serves as
+a default when the iterable is empty.
+
+For example, reduce(lambda x, y: x+y, [1, 2, 3, 4, 5])
+calculates ((((1 + 2) + 3) + 4) + 5).
```

Comment:
Let's keep the docstring as it was. If it needs amendment, we should do so in a separate PR, because _that_ PR would then be backported to all active branches (though, of course backported manually bco. the Argument Clinic changes).

Reply:
Unfortunately, AC requires a summary line.  The reduce() docstring doesn't fit in that format as-is.

Reply:
Right, so let's amend the docstring first (in a separate PR).

Reply:
The docstring change here seems small enough that it's fine to do it also in this PR.

Reply:
FYI: https://github.com/python/cpython/pull/126045

Reply:
pr updated to include docstrings changes.